admin/app/controllers/R2PressController.scala

```diff
@@ -0,0 +1,86 @@
+package controllers.admin
+
+import java.io.File
+
+import common.{ExecutionContexts, Logging}
+import controllers.AuthLogging
+import play.api.mvc.{AnyContent, Controller}
+import services.{R2PressedPageTakedownNotifier, R2PagePressNotifier}
+
+case class R2PagePress(r2url: String) {
+  def trim = this.copy(r2url = r2url.trim)
+}
+
+object R2PressController extends Controller with Logging with AuthLogging with ExecutionContexts {
+
+  def pressForm() = AuthActions.AuthActionTest { implicit request =>
+    Ok(views.html.pressR2())
+  }
+
+  def batchUpload() = AuthActions.AuthActionTest { implicit request =>
+    val body = request.body
+    val uploadedFile = body.asMultipartFormData.flatMap { files =>
+      files.file("r2urlfile").map { theFile =>
+        val rnd = Math.random().toString.replace(".","")
+        val tmpName = s"/tmp/$rnd${theFile.filename}"
+        val tmpFile = new File(tmpName)
+        theFile.ref.moveTo(tmpFile)
+        tmpFile
+      }
+    }
+    uploadedFile.foreach(file => pressFile(file, isBatchTakedown(body)))
+    Ok(views.html.pressR2())
```

Comment:
The pressing happens asynchronously so we can't wait for the file contents to be processed but maybe It'll be worth adding a test for uploadedFile.nonEmpty though.